sklego/meta/grouped_predictor.py

```diff
@@ -292,9 +294,15 @@ def __predict_single_group(self, group, X, method="predict"):
                     f"Found new group {group} during predict with use_global_model = False"
                 )
 
+        # Ensure to mantain label order if it's called with predict_proba
+        if is_classifier(group_predictor) and method == "predict_proba":
+            extra_kwargs = {"columns": group_predictor.classes_}
+
```

Comment:
Couple of nitpicks:
1. The comment may be misleading, the order is actually maintained by the behaviour of [pd.concat](https://pandas.pydata.org/docs/reference/api/pandas.concat.html) (with `axis=0`), here we just register the column names (labels) that appear in the specific group
2. Probably personal preference but I would opt for something along the line:
    ```py
    if condition:
        extra_kwargs = {"columns": group_predictor.classes_}
    else:
        extra_kwargs = {}
    ```
    where you don't initialize the extra_kwargs at the beginning of the function